pandas/core/frame.py

```diff
@@ -7390,7 +7391,9 @@ def f(x):
             return op(x, axis=axis, skipna=skipna, **kwds)
 
         # exclude timedelta/datetime unless we are uniform types
-        if axis == 1 and self._is_mixed_type and self._is_datelike_mixed_type:
+        if (axis == 1 and self._is_datelike_mixed_type
+                and (not self._is_homogeneous_type
+                     and not is_datetime64tz_dtype(self.dtypes[0]))):
```

Comment:
For those wondering, `DataFrame. _is_datelike_mixed_type` is False for an empty DataFrame, so we don't have to worry about an IndexError when getting `.dtypes[0]`, since we should have short-circuited.

(`._is_homogeneous_type` is true for an empty DataFrame, which I suppose makes sense)